src/main/scala/mesosphere/marathon/api/akkahttp/v2/LeaderController.scala

```diff
@@ -0,0 +1,70 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.http.scaladsl.server.Route
+import mesosphere.marathon.api.akkahttp.{ Controller, Rejections }
+import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.plugin.auth._
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.raml.RuntimeConfiguration
+import mesosphere.marathon.storage.repository.RuntimeConfigurationRepository
+
+import scala.async.Async._
+import scala.concurrent.ExecutionContext
+
+case class LeaderController(
+    val electionService: ElectionService,
+    val runtimeConfigRepo: RuntimeConfigurationRepository)(
+    implicit
+    val authenticator: Authenticator,
+    val authorizer: Authorizer,
+    val executionContext: ExecutionContext) extends Controller with StrictLogging {
+
+  import mesosphere.marathon.api.akkahttp.Directives._
+  import mesosphere.marathon.api.akkahttp.EntityMarshallers._
+
+  def leaderInfo(): Route =
+    authenticated.apply { implicit identity =>
+      authorized(ViewResource, AuthorizedResource.SystemConfig).apply {
+        electionService.leaderHostPort match {
+          case None => reject(Rejections.EntityNotFound.leader())
+          case Some(leader) => complete(raml.LeaderInfo(leader))
+        }
+      }
+    }
+
+  def deleteLeader(): Route =
+    authenticated.apply { implicit identity =>
+      authorized(UpdateResource, AuthorizedResource.SystemConfig).apply {
+        parameters('backup.?, 'restore.?) { (backup, restore) =>
+          // This follows the LeaderResource implementation. Seems like a bug to me. Why should we not proxy the request?
```

Comment:
I'd say this should be an `asLeader`. What do you say @meln1k and @timcharper?

Reply:
I think there should be some kind of directive for that. But I'm not sure if it's merged yet.

Reply:
`asLeader` should work fine here. It will also proxy the request.

Reply:
Yep. Proxying most of the requests should probably happen elsewhere. Otherwise the code below is only executed `asLeader`

Reply:
Can we place `asLeader` in the most upper level of the routes structure?

Reply:
Ok. It seems the original resource has a bug then.

Reply:
I am not sure if that's actually a bug. May be `DELETE /leader` was not supposed to be proxied for some reason?